### PR TITLE
Miscellaneous improvements for regexmanager

### PR DESCRIPTION
**Pull request by dennisschagt**

### CODE REVIEW

**Comment by dennisschagt** [comment]
@Minoru: Sorry for the multiple force-pushes. I didn't like [my original approach](https://github.com/newsboat/newsboat/compare/863864446edc4afec0ef9947c53c230ba202767d...dennisschagt:tmp/regex-class-backup). It is ready for review now.

**Comment by Minoru** [comment]
> Sorry for the multiple force-pushes.

It's fine as long as it isn't in the middle of a review. I already got used to you requesting the review via GitHub, so when I saw your PR last night, I simply ignored it because I knew you're still working on it.

Also, thanks for going around the codebase doing refactorings like this!

File: include/regexmanager.h [hunk 1]
```diff
@@ -41,10 +42,7 @@ class RegexManager : public ConfigActionHandler {
 		const std::vector<std::string>& params);
 
 public:
-	std::vector<std::string>& get_attrs(const std::string& loc)
-	{
-		return locations[loc].second;
-	}
+	std::string get_attrs_stfl_string(const std::string& location, bool hasFocus);
```
[the remark this answers is not in the document]
Comment (dennisschagt) [reply include/regexmanager.h]: Good idea, done

**Comment by dennisschagt** [comment]
I've made the changes requested in the review comments.
For now, I've only kept the first 2 commits. I think it is better if I create a separate PR for the other changes so this can be merged if you are ok with it.

image

**Comment by dennisschagt** [comment]
> The first two commits look fine to me. I'll merge them if you address my few minor comments.

Sounds good, I've removed the other commits from this branch.

> As for the wrapper: I think I understand where you're coming from — you're trying to automate memory management without screwing up the C API

Correct, I just wanted to get rid of the multiple copies of `regfree(x); delete x;`.
 
> Why did you move away from your original approach?

Multiple reasons:
- I don't like the required copy assignment and copy constructor implementation as it is easy to introduce subtle issues.
- In the copy assignment/constructor, `compile()` was called without checking for errors. Error handling in the copy constructor seems only possible with exceptions which I don't want to introduce due to https://github.com/newsboat/newsboat/pull/882#pullrequestreview-391894876.

>  I think making the wrapper closer to regex-rs would have been the way to go, since it would _both_ automate memory management and make it trivial to port `RegexManger` to Rust. 

Agreed, I have some idea and will create a PR for that if it works out to my liking.

**Pull request merged**